### PR TITLE
MAISTRA-1627, MAISTRA-1660: Fix lint, gen-check and integration test failures

### PR DESCRIPTION
**Pull request by jwendell**
This PR covers a bunch of fixes for failures:
- Lint failures: now `make lint` works (MAISTRA-1627)
- Gen failures: now `make gen-check` works (MAISTRA-1627)
- Integration test failures: now integration tests work (MAISTRA-1660)

For the last item we had to bring the fix from upstream release-1.6 branch, hence some commits not related to this PR itself. This is to avoid cherry-picking just one commit and have to resolve conflicts later.

### CODE REVIEW

**Review by knrc — APPROVED** [review]

File: pkg/listwatch/listwatch_test.go [hunk 1]
```diff
@@ -456,7 +456,7 @@ func TestMultiWatchDoesNotDeadlock(t *testing.T) {
 			},
 			WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
 				if namespace == "4" {
-					return nil, errors.New("Some error occurred")
+					return nil, errors.New("some error occurred")
```
Comment (brian-avery) [inline pkg/listwatch/listwatch_test.go:459]: is there a reason we don't return the error? Also, would it be better to say "an error occurred"?

File: pkg/servicemesh/apis/servicemesh/v1/doc.go [hunk 2]
```diff
@@ -1,3 +1,17 @@
+// Copyright Istio Authors
```
Comment (brian-avery) [inline pkg/servicemesh/apis/servicemesh/v1/doc.go]: @knrc who owns the copyright for code that we write in our fork of Istio? Files that we add? Should this be copyright Istio or copyright Red Hat? Or even copyright Maistra?

**Review by brian-avery — CHANGES_REQUESTED** [review]
Leaving this open until questions are answered.

**Review by knrc — CHANGES_REQUESTED** [review]
Holding until we have Brian's question answered

File: pkg/listwatch/listwatch_test.go (hunk 1 above)
Comment (knrc) [reply pkg/listwatch/listwatch_test.go:459]: If this is not code we have authored then we should keep it matching upstream, the only possible link error would be the capitalisation.

Comment (brian-avery) [reply pkg/listwatch/listwatch_test.go:459]: oh. Just realized this is in test code anyway. But I think we added the listwatcher for the multitenancy work didn't we?

Comment (knrc) [reply pkg/listwatch/listwatch_test.go:459]: No, they had already copied a version of it from the prometheus codebase

**Review by dgn — APPROVED** [review]

File: pkg/servicemesh/apis/servicemesh/v1/doc.go (hunk 2 above)
Comment (knrc) [reply pkg/servicemesh/apis/servicemesh/v1/doc.go]: This does need to change, the answer is that we should not be using "Copyright Istio Authors" unless it's part of the upstream

**Comment by jwendell** [comment]
Fixed the Copyright

**Comment by jwendell** [comment]
All tests pass now. Turns out integration tests had an issue. Created MAISTRA-1660 for that.
Pulled in the fix for it from upstream 1.6 branch in this PR so that all tests pass and we have a single PR fixing all test/lint failures.

**Comment by knrc** [comment]
@jwendell This PR seems to include more than just lint and gen-check failures, can these be separated out into individual PRs?

**Comment by jwendell** [comment]
no. If we split it, tests will fail for both PR's. Unless we force-merge them.

**Comment by knrc** [comment]
@jwendell  please create a second JIRA to cover the additional work and include it in the description of this PR, we should not be sneaking in additional changes just for expedience.

**Review by knrc — CHANGES_REQUESTED** [review]
Requesting changes to cover additional commits included within this PR

**Comment by jwendell** [comment]
@knrc done

**Review by knrc — APPROVED** [review]
@jwendell thanks very much

**Review by brian-avery — APPROVED** [review]

**Pull request merged**